### PR TITLE
Fixes a runtime when flipping a welding mask as a non-human

### LINKED ISSUE #10671: Flipping a welding hat while being a mob that cannot wear it runtimes
### Labels

[BUG]

### Description

If a silicon were to obtain a welding helmet (such as by using a magtractor as a ghostdrone) and use the welding helmet in-hand, the game runtimes (line 692 of code\obj\item\clothing\helmets.dm)

### Steps to reproduce

1: Be a ghostdrone with a magtractor or otherwise be able to pick things up as a silicon.
2: Pick up a welding helmet
3: Attempt to flip it to the hat position
4: Game runtimes

### Expected Behavior

You shouldnt need to be able to wear a helmet to flip it.

### Additional Information & Screenshots

_No response_

### PR DESCRIPTION
**Pull request by Bartimeus973**
[C-bug]
## About the PR 
Adds a "ishuman()" check before checking for a head to fix a runtime happening when you flip a mask as someone without a head (like a ghostdrone)

## Why's this needed? 
Bug bad, i fix
Fixes #10671

### CODE REVIEW

File: code/obj/item/clothing/helmets.dm [hunk 1]
```diff
@@ -684,7 +684,7 @@
 		see_face = FALSE
 		icon_state = "welding"
 		boutput(user, "You flip the mask down. The mask is now protecting you from eye damage.")
-		if (user.head == src)
+		if (ishuman(user) && (user.head == src))
```
Comment (ZeWaka) [inline code/obj/item/clothing/helmets.dm:687]: this won't update the sprite properly - why allow nonhumans to flip them at all

**Review by ZeWaka — CHANGES_REQUESTED** [review]
this is just a bandaid - how do nonhumans even call this proc

**Comment by Bartimeus973** [comment]
Changed the ishuman() check to be used on attack_self(), before the proc call

File: code/obj/item/clothing/helmets.dm [hunk 2]
```diff
@@ -722,8 +725,9 @@
 				src.reveal(owner)
 
 	attack_self(mob/user) //let people toggle these inhand too
-		for(var/obj/ability_button/mask_toggle/toggle in ability_buttons)
-			toggle.execute_ability() //This is a weird way of doing it but we'd have to get the ability button to update the icon anyhow
+		if (ishuman(user))
+			for(var/obj/ability_button/mask_toggle/toggle in ability_buttons)
+				toggle.execute_ability() //This is a weird way of doing it but we'd have to get the ability button to update the icon anyhow
```
Comment (ZeWaka) [inline code/obj/item/clothing/helmets.dm:730]: but why do they even have the ability button?

Comment (ZeWaka) [reply code/obj/item/clothing/helmets.dm]: x

Comment (Bartimeus973) [reply code/obj/item/clothing/helmets.dm]: I've hopefully hidden the button.

**Pull request merged**